lms/static/js/edxnotes/utils/notes_collector.js

```diff
@@ -0,0 +1,87 @@
+(function(define) {
+    'use strict';
+    define(['jquery', 'underscore', 'annotator_1.2.9'], function($, _, Annotator) {
+        var cleanup,
+            renderNotes,
+            fetchNotesWhenReady,
+            storeNotesRequestData,
+            searchRequestsData = [];
+
+        cleanup = function () {
+            searchRequestsData = [];
+        };
+
+        storeNotesRequestData = function (data) {
+            searchRequestsData.push(data);
+            fetchNotesWhenReady();
+        };
+
+        fetchNotesWhenReady = function () {
+            if ($('.edx-notes-wrapper').length != searchRequestsData.length) {
```

Comment:
oh..must have missed this. We can move it into `notes_factory.js` but then we have to pass the expected number of notes in `storeNotesRequestData` call every time. Do you think it makes sense?